common/controlplane/v1/upgrade.go

```diff
@@ -40,8 +47,16 @@ func (cp CPv1) Upgrade(isUpgradingToUnstableBranch, isPartialRebuildDisableNeede
 
 	// Append arguments based on conditions
 	if isUpgradingToUnstableBranch {
-		cmdArgs = append(cmdArgs, "--registry", CIRegistry, string(AllowUpgradeToUnstableBranchFlag))
+		// for plugin version with rc tag i.e. release candidate
+		// upgrade job images are not present in ci-registry, they are in
+		// docker hub. so for those tags we dont need to add --registry flag.
+		if strings.Contains(pluginVersion, "rc") {
```

Comment:
maybe make this string a bit more specific like '-rc.' ?

Reply:
Would it be simpler to have a helper function which return the correct cmdArgs based on pluginVersion?

Reply:
updated it.